src/inspect.rs

```diff
@@ -189,3 +209,106 @@ impl MetadataPrinter for MetadataPrettyPrinter {
         self.print_metadata_usage(metadata)
     }
 }
+
+fn get_signatures_printer(output_type: OutputType) -> Box<dyn SignaturesPrinter> {
+    match output_type {
+        OutputType::Yaml => Box::new(SignaturesYamlPrinter {}),
+        OutputType::Pretty => Box::new(SignaturesPrettyPrinter {}),
+    }
+}
+
+trait SignaturesPrinter {
+    fn print(&self, signatures: &OciImageManifest);
+}
+
+struct SignaturesPrettyPrinter {}
+
+impl SignaturesPrinter for SignaturesPrettyPrinter {
+    fn print(&self, signatures: &OciImageManifest) {
+        for layer in &signatures.layers {
+            let mut table = Table::new();
+            table.set_format(FormatBuilder::new().padding(0, 1).build());
+            table.add_row(row![Fmbl -> "Digest: ", layer.digest]);
+            table.add_row(row![Fmbl -> "Media type: ", layer.media_type]);
+            table.add_row(row![Fmbl -> "Size: ", layer.size]);
+            if let Some(annotations) = &layer.annotations {
+                table.add_row(row![Fmbl -> "Annotations"]);
+                for annotation in annotations.iter() {
+                    table.add_row(row![Fgbl -> annotation.0, annotation.1]);
+                }
+            }
+            table.printstd();
+            println!();
+        }
+    }
+}
+
+struct SignaturesYamlPrinter {}
+
+impl SignaturesPrinter for SignaturesYamlPrinter {
+    fn print(&self, signatures: &OciImageManifest) {
+        let signatures_yaml = serde_yaml::to_string(signatures);
+        if let Ok(signatures_yaml) = signatures_yaml {
+            println!("{}", signatures_yaml)
+        }
+    }
+}
+
+async fn fetch_signatures_manifest(
+    uri: &str,
+    sources: Option<Sources>,
+    docker_config: Option<DockerConfig>,
+) -> Option<OciImageManifest> {
+    let registry = Registry::new(docker_config.as_ref());
+    let digest = registry.manifest_digest(uri, sources.as_ref()).await.ok()?;
+    let signature_url = get_signature_url(uri.to_string(), digest.as_str())?;
+    let manifest = registry
+        .manifest(signature_url.as_str(), sources.as_ref())
+        .await
+        .ok()?;
+
+    match manifest {
+        OciManifest::Image(img) => Some(img),
+        _ => None,
+    }
+}
+
+// simulate cosign triangulate output. Use the manifest digest, which returns the sha256 of the manifest content, then replace ':' with '-' and append '.sig'
+fn get_signature_url(mut uri: String, digest: &str) -> Option<String> {
```

Comment:
Why do we have to simulate the triangulate function of sigstore? I'm pretty sure `policy-fetcher` is already exposing sigstore, hence we can just use the original function.

If `policy-fetcher` is not exposing sigstore, we can change it